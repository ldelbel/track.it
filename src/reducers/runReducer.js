import { ADD_RUNNING_SESSION, SET_USER, FILL_LIST } from '../actions';

let initialState = {
  id: 0,
  username: '',
  runningSessions: [],
};

if (localStorage.user) {
  initialState = {
    id: localStorage.id,
    username: localStorage.user,
    runningSessions: JSON.parse(localStorage.runningSessions),
  };
}

const runReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_RUNNING_SESSION:
      return {
        ...state,
        runningSessions: state.runningSessions.concat(action.payload),
      };
    case SET_USER:
      return {
        ...state,
        username: action.payload,
      };
    case FILL_LIST:
      return {
        ...state,
        id: action.payload[0],
        runningSessions: action.payload[1],
      };
    default:
      return state;
  }
};

export default runReducer;
