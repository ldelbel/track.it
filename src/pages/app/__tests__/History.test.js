import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import History from '../history';
import runReducer from '../../../reducers/runReducer';

describe('history component',() => {
    let running_sessions = [
        {
            id: 1,
            user: 'test1',
            distance: 1,
            duration: 1,
            avg_speed: 1,
            avg_pace: 1,
            goal: 1,
        },{
            id: 2,
            user: 'test2',
            distance: 2,
            duration: 2,
            avg_speed: 2,
            avg_pace: 2,
            goal: 2,
        }
    ]

    it('renders correctly', () => {
        const store = createStore(runReducer);
        const history = renderer.create(
          <Provider store={store}>
            <History runningSessions={running_sessions} />
          </Provider>,
        ).toJSON();
        expect(history).toMatchSnapshot();
    });
})