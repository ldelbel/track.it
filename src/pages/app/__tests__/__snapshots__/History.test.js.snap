// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history component renders correctly 1`] = `
<main
  className="bg"
>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <div
        className="container__leftdiv"
      >
        <div
          className="container__leftdiv__progressbar"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <circle
              className="CircularProgressbar-background"
              cx={50}
              cy={50}
              r={50}
              style={
                Object {
                  "fill": "#fff",
                }
              }
            />
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-43
      a 43,43 0 1 1 0,86
      a 43,43 0 1 1 0,-86
    "
              fillOpacity={0}
              strokeWidth="14"
              style={
                Object {
                  "stroke": "#f3f3f3",
                  "strokeDasharray": "270.1769682087222px 270.1769682087222px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-43
      a 43,43 0 1 1 0,86
      a 43,43 0 1 1 0,-86
    "
              fillOpacity={0}
              strokeWidth="14"
              style={
                Object {
                  "stroke": "var(--green)",
                  "strokeDasharray": "270.1769682087222px 270.1769682087222px",
                  "strokeDashoffset": "0px",
                }
              }
            />
          </svg>
        </div>
        <div
          className="container__leftdiv__info"
        >
          <span>
            Invalid Date
          </span>
          <p>
            100.00
            % of the goal
          </p>
        </div>
      </div>
      <div
        className="container__rightdiv"
      >
        <div
          className="container__rightdiv__info"
        >
          <span>
            1
          </span>
          <p>
            km
          </p>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="container__leftdiv"
      >
        <div
          className="container__leftdiv__progressbar"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <circle
              className="CircularProgressbar-background"
              cx={50}
              cy={50}
              r={50}
              style={
                Object {
                  "fill": "#fff",
                }
              }
            />
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-43
      a 43,43 0 1 1 0,86
      a 43,43 0 1 1 0,-86
    "
              fillOpacity={0}
              strokeWidth="14"
              style={
                Object {
                  "stroke": "#f3f3f3",
                  "strokeDasharray": "270.1769682087222px 270.1769682087222px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-43
      a 43,43 0 1 1 0,86
      a 43,43 0 1 1 0,-86
    "
              fillOpacity={0}
              strokeWidth="14"
              style={
                Object {
                  "stroke": "var(--green)",
                  "strokeDasharray": "270.1769682087222px 270.1769682087222px",
                  "strokeDashoffset": "0px",
                }
              }
            />
          </svg>
        </div>
        <div
          className="container__leftdiv__info"
        >
          <span>
            Invalid Date
          </span>
          <p>
            100.00
            % of the goal
          </p>
        </div>
      </div>
      <div
        className="container__rightdiv"
      >
        <div
          className="container__rightdiv__info"
        >
          <span>
            2
          </span>
          <p>
            km
          </p>
        </div>
      </div>
    </div>
  </div>
</main>
`;
